pkg/controller/nodeconfig/sync.go

```diff
@@ -73,6 +78,11 @@ func (ncc *Controller) sync(ctx context.Context, key string) error {
 		objectErrs = append(objectErrs, err)
 	}
 
+	roleBindings, err := ncc.getRoleBindings()
```

Comment:
This file already has a mixed way of getting dependent objects, lets use new approach for new code.

Reply:
doesn't `controllerhelpers.GetObjects` only apply to controlled objects, while these are non-namespaced? Unless you mean something else by new approach

Reply:
They are originating from NodeConfig so there should be a controllerRef on them, and these are namespaced RoleBindings and Roles.

Reply:
the existing SA doesn't use controller ref either

https://github.com/scylladb/scylla-operator/blob/11394625d288542782cd23fee50a3d9ff6957ebe/pkg/controller/nodeconfig/resource.go#L29-L39
These are shared by multiple NodeConfig and a for the lack of a better place we reconcile them with each of them.

Reply:
Then enqueueing owner in event handlers must not work and your PR does exactly that

Reply:
yeah, that should enqueueAll but it must be broken in all the other places already there that I copied it from

Reply:
updated based on #2000